File: 7term/web/lab5/src/entity/Employee.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToOne, JoinColumn } from "typeorm";
import { Department } from "./Department";
import { ComputerSpecs } from "./ComputerSpecs";

@Entity()
export class Employee {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column({ unique: true })
  email: string;

  @ManyToOne(() => Department, (department) => department.employees)
  department: Department;

  @OneToOne(() => ComputerSpecs, { cascade: true, onDelete: "CASCADE" })
  @JoinColumn()
  computerSpecs: ComputerSpecs;
}

File: 7term/web/lab6/server/src/app/models/employee.py
from sqlalchemy import Column, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from ..core.db.database import Base


class Employee(Base):
    __tablename__ = "employee"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    email = Column(String, unique=True, nullable=False)
    department_id = Column(Integer, ForeignKey("department.id"), nullable=False)
    computer_specs_id = Column(Integer, ForeignKey("computer_specs.id"), nullable=False)

    department = relationship("Department", back_populates="employees")
    computer_specs = relationship("ComputerSpecs", cascade="all, delete")
